Regression in 8.4: Failed to build variable mapping expression

**Describe the bug**

I can't deploy a BPMN process to Zeebe 8.4.0. In previous versions (8.3.5), I deployed the process successfully. The deployment reports the following error:

```
Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'New BPMN Diagram.bpmn': Failed to build variable mapping expression: failed to parse expression '{apiKey:key,from:{name:"Me",email:"me@camunda.com"},to:{name:""You"",email:"you@camunda.com"},unMappedFieldNotUseInModel:{mailType:"mail"},content:{subject:"Hello",type:"text/plain",value:"This is a test."}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:ke"[ deploy-error ] 23/1/2024 8:37:14
``` 

![Screenshot from 2024-01-23 08-40-09](https://github.com/camunda/zeebe/assets/4305769/75e8e85a-c8ad-4689-a84b-53f6e246f068)

The process contains a SendGrid connector task with the configuration:
- Receiver Name = `"You"` (static value but with double-quotes)

**To Reproduce**

1. Deploy the BPMN process [regression-variable-mapping.bpmn](https://github.com/camunda/zeebe/files/14019955/regression-variable-mapping.bpmn.txt)
2. Verify that the deployment fails

**Expected behavior**

I can deploy the BPMN process.

If the static value is wrapped in double quotes then the double quotes are removed on parsing. When the variable mapping is applied, the process variable should be a string value `You` without double quotes (i.e. as JSON `"You"` and not `"\"You\""`).

**Log/Stacktrace**

This issue was reported as a [Support case](https://jira.camunda.com/browse/SUPPORT-20145). 

**Environment:**
- OS: <!-- [e.g. Linux] -->
- Zeebe Version: `8.4.0`
- Configuration: <!-- [e.g. exporters etc.] -->

@saig0 @abbasadel as written [here](https://camunda.slack.com/archives/C06EFPJ076W/p1706016589227629?thread_ts=1706012506.189009&cid=C06EFPJ076W) I think this issue is more problematic then we thought. I was able to reproduce this with a simple update test.


## Update Test

Setting up Camunda 8 SM (our benchmark charts) with following values:

- [values.83.txt](https://github.com/camunda/zeebe/files/14024840/values.83.txt)
- [values.84.txt](https://github.com/camunda/zeebe/files/14024842/values.84.txt)

Values for 8.3 I use to bootstrap the cluster. Then I use the model shared by @saig0 to deploy the process:

```
$ zbctl deploy --insecure /tmp/regression-variable-mapping.bpmn
{
  "key": "2251799813685250",
  "processes": [
    {
      "bpmnProcessId": "Process_0zc7ixs",
      "version": 1,
      "processDefinitionKey": "2251799813685249",
      "resourceName": "/tmp/regression-variable-mapping.bpmn"
    }
  ]
}
```

I create some instances (without instances it doesn't fail).

```
[cqjawa ck-update-test/ cluster: zeebe-cluster ns:ck-update-test]$ zbctl create instance 2251799813685249 --insecure
{
  "processDefinitionKey": "2251799813685249",
  "bpmnProcessId": "Process_0zc7ixs",
  "version": 1,
  "processInstanceKey": "2251799813685251"
}
[cqjawa ck-update-test/ cluster: zeebe-cluster ns:ck-update-test]$ zbctl create instance 2251799813685249 --insecure
{
  "processDefinitionKey": "2251799813685249",
  "bpmnProcessId": "Process_0zc7ixs",
  "version": 1,
  "processInstanceKey": "4503599627370497"
}
[cqjawa ck-update-test/ cluster: zeebe-cluster ns:ck-update-test]$ zbctl create instance 2251799813685249 --insecure
{
  "processDefinitionKey": "2251799813685249",
  "bpmnProcessId": "Process_0zc7ixs",
  "version": 1,
  "processInstanceKey": "6755399441055745"
}
```

Afterwards I do the upgrade to 8.4

```
cp values.84.txt values.yaml
make update
helm upgrade --namespace ck-update-test ck-update-test zeebe-benchmark/zeebe-benchmark --reuse-values -f values.yaml
W0123 14:47:49.188447   40837 warnings.go:70] spec.template.spec.containers[0].env[16].name: duplicate name "ZEEBE_LOG_LEVEL"
W0123 14:47:49.950730   40837 warnings.go:70] spec.template.spec.containers[0].env[25].name: duplicate name "K8S_NAMESPACE"
W0123 14:47:49.950745   40837 warnings.go:70] spec.template.spec.containers[0].env[26].name: duplicate name "K8S_NAME"
W0123 14:47:49.950750   40837 warnings.go:70] spec.template.spec.containers[0].env[31].name: duplicate name "ZEEBE_LOG_LEVEL"
Release "ck-update-test" has been upgraded. Happy Helming!
NAME: ck-update-test
LAST DEPLOYED: Tue Jan 23 14:47:46 2024
NAMESPACE: ck-update-test
STATUS: deployed
REVISION: 2
NOTES:
# Zeebe Benchmark

Installed Zeebe cluster with:

 * 3 Brokers
 * 2 Gateways

The benchmark is running with:

 * Starter replicas=0
 * Worker replicas=0
 * Publisher replicas=0
 * Timer replicas=0

```

We can see that the nodes come up, but are unhealthy. 
![unhealthy-reproduction](https://github.com/camunda/zeebe/assets/2758593/7f7222cc-7e42-46c8-8333-e0cdef95a8bf)


[Logs indicating](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22ck-update-test%22%0Alabels.k8s-pod%2Fapp%3D%22camunda-platform%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22zeebe-broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22ck-update-test%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22camunda-platform%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fpart-of%3D%22camunda-platform%22;pinnedLogId=2024-01-23T13:49:17.718985305Z%2F0cukdcxfj53ua6rk;summaryFields=labels%252F%2522k8s-pod%252Fstatefulset_kubernetes_io%252Fpod-name%2522:false:32:beginning;cursorTimestamp=2024-01-23T13:49:17.718985305Z;duration=PT5M?project=zeebe-io) the same issue:


```
java.lang.RuntimeException: java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{apiKey:key,from:{name:"Me",email:"me@camunda.com"},to:{name:""You"",email:"you@camunda.com"},unMappedFieldNotUseInModel:{mailType:"mail"},content:{subject:"Hello",type:"text/plain",value:"This is a test."}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:ke"
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.lambda$replayNextEvent$4(ReplayStateMachine.java:168) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) ~[zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:96) ~[zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:120) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.4.0.jar:8.4.0]
Caused by: java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{apiKey:key,from:{name:"Me",email:"me@camunda.com"},to:{name:""You"",email:"you@camunda.com"},unMappedFieldNotUseInModel:{mailType:"mail"},content:{subject:"Hello",type:"text/plain",value:"This is a test."}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:ke"
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.parseExpression(VariableMappingTransformer.java:196) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.transformInputMappings(VariableMappingTransformer.java:84) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.lambda$transformIoMappings$1(FlowNodeTransformer.java:64) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at java.base/java.util.Optional.map(Unknown Source) ~[?:?]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transformIoMappings(FlowNodeTransformer.java:62) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform(FlowNodeTransformer.java:38) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform(FlowNodeTransformer.java:21) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.TransformationVisitor.visit(TransformationVisitor.java:42) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.model.bpmn.traversal.TypeHierarchyVisitor.visit(TypeHierarchyVisitor.java:39) ~[zeebe-bpmn-model-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.model.bpmn.traversal.ModelWalker.walk(ModelWalker.java:73) ~[zeebe-bpmn-model-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.BpmnTransformer.transformDefinitions(BpmnTransformer.java:118) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.updateInMemoryState(DbProcessState.java:234) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.lookupPersistenceStateForProcessByKey(DbProcessState.java:438) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.getProcessByKeyAndTenant(DbProcessState.java:328) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.getFlowElement(DbProcessState.java:364) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.createEventScope(ProcessInstanceElementActivatingApplier.java:267) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:51) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.appliers.ProcessInstanceElementActivatingApplier.applyState(ProcessInstanceElementActivatingApplier.java:32) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.appliers.EventAppliers.applyState(EventAppliers.java:356) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.Engine.replay(Engine.java:95) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.lambda$replayEvent$7(ReplayStateMachine.java:231) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at java.base/java.util.Optional.ifPresent(Unknown Source) ~[?:?]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.replayEvent(ReplayStateMachine.java:231) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at java.base/java.util.Iterator.forEachRemaining(Unknown Source) ~[?:?]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.lambda$tryToReplayBatch$5(ReplayStateMachine.java:207) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:98) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.tryToReplayBatch(ReplayStateMachine.java:205) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.lambda$replayNextEvent$3(ReplayStateMachine.java:164) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.retry.ActorRetryMechanism.run(ActorRetryMechanism.java:28) ~[zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.retry.RecoverableRetryStrategy.run(RecoverableRetryStrategy.java:48) ~[zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) ~[zeebe-scheduler-8.4.0.jar:8.4.0]
	... 5 more

```

## Conclusion

This problem will arise if the customer has deployed a process model with double quotes (?) and has running instances with it. 

Since we don't support down-grading (due to data migration etc.) I see this issue as critical as there is no work-around. Let me know if you see one.

--- a/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java
@@ -154,7 +154,10 @@ public final class VariableMappingTransformer {
         final String expression;
 
         if (sourceExpression instanceof StaticExpression) {
-          expression = String.format("\"%s\"", sourceExpression.getExpression());
+          // due to a regression (https://github.com/camunda/zeebe/issues/16043) all the double
+          // quotes inside the static expression must be escaped
+          expression =
+              String.format("\"%s\"", sourceExpression.getExpression().replaceAll("\"", "\\\\\""));
         } else {
           expression = sourceExpression.getExpression();
         }

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java
@@ -78,7 +78,9 @@ public final class ActivityTest {
                       .zeebeInput("https://github.com/{{orgId}}/{{repoId}}", "urlStatic")
                       .zeebeInputExpression(
                           "\"https://github.com/{{orgId}}/{{repoId}}\"", "urlExpression")
-                      .zeebeInput("My Name is &#34;Zeebe&#34;, nice to meet you", "quotesStatic")
+                      .zeebeInput(
+                          "My Name is &#34;Zeebe&#34;, nice to meet you", "quotesStaticModeler")
+                      .zeebeInput("My Name is \"Zeebe\", nice to meet you", "quotesStatic")
                       .zeebeInputExpression(
                           "\"My Name is \\\"Zeebe\\\", nice to meet you\"", "quotesExpression"))
           .endEvent()
@@ -143,7 +145,8 @@ public final class ActivityTest {
             entry("nullExpression", "\"null\""),
             entry("urlStatic", "\"https://github.com/{{orgId}}/{{repoId}}\""),
             entry("urlExpression", "\"https://github.com/{{orgId}}/{{repoId}}\""),
-            entry("quotesStatic", "\"My Name is &#34;Zeebe&#34;, nice to meet you\""),
+            entry("quotesStaticModeler", "\"My Name is &#34;Zeebe&#34;, nice to meet you\""),
+            entry("quotesStatic", "\"My Name is \\\"Zeebe\\\", nice to meet you\""),
             entry("quotesExpression", "\"My Name is \\\"Zeebe\\\", nice to meet you\""));
   }
 
@@ -226,6 +229,41 @@ public final class ActivityTest {
   @Test
   public void shouldIgnoreTaskHeadersIfNull() {
     createProcessAndAssertIgnoredHeaders(null);
+  }
+
+  // regression test for https://github.com/camunda/zeebe/issues/16043
+  @Test
+  public void shouldEscapeDoubleQuotesInStaticExpression() {
+    final var process = createModelFromClasspathResource("/regression-variable-mapping.bpmn");
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId("regressionProcess").create();
+
+    final Record<ProcessInstanceRecordValue> record =
+        RecordingExporter.processInstanceRecords()
+            .withElementId("send-email")
+            .withIntent(ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> variables =
+        ProcessInstances.getCurrentVariables(processInstanceKey, record.getPosition());
+
+    assertThat(variables)
+        .contains(
+            entry("unMappedFieldNotUseInModel", "{\"mailType\":\"mail\"}"),
+            entry("apiKey", "null"),
+            entry("from", "{\"name\":\"Me\",\"email\":\"me@camunda.com\"}"),
+            entry("to", "{\"name\":\"\\\"You\\\"\",\"email\":\"you@camunda.com\"}"),
+            entry(
+                "content",
+                "{\"subject\":\"Hello\",\"type\":\"text/plain\",\"value\":\"This \\\"is a\\\" test.\"}"));
+  }
+
+  private BpmnModelInstance createModelFromClasspathResource(final String classpath) {
+    final var resourceAsStream = getClass().getResourceAsStream(classpath);
+    return Bpmn.readModelFromStream(resourceAsStream);
   }
 
   private void createProcessAndAssertIgnoredHeaders(final String testValue) {

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java
@@ -106,6 +106,7 @@ public final class VariableMappingTransformerTest {
                 mapping("Hello\tWorld", "tab"),
                 mapping("Hello\nWorld", "newline"),
                 mapping("Hello\rWorld", "carriageReturn"),
+                mapping("\"My Name is \"Zeebe\", nice to meet you\"", "doubleQoutes"),
                 mapping("My Name is &#34;Zeebe&#34;, nice to meet you", "encodedQuotes")),
             expressionLanguage);
 
@@ -115,7 +116,7 @@ public final class VariableMappingTransformerTest {
         .isTrue();
     assertThat(expression.getExpression())
         .isEqualTo(
-            "{tab:\"Hello\tWorld\",newline:\"Hello\nWorld\",carriageReturn:\"Hello\rWorld\",encodedQuotes:\"My Name is &#34;Zeebe&#34;, nice to meet you\"}");
+            "{tab:\"Hello\tWorld\",newline:\"Hello\nWorld\",carriageReturn:\"Hello\rWorld\",doubleQoutes:\"\\\"My Name is \\\"Zeebe\\\", nice to meet you\\\"\",encodedQuotes:\"My Name is &#34;Zeebe&#34;, nice to meet you\"}");
   }
 
   private static ZeebeMapping mapping(final String source, final String target) {

--- a/engine/src/test/resources/regression-variable-mapping.bpmn
+++ b/engine/src/test/resources/regression-variable-mapping.bpmn
@@ -1,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:zeebe="http://camunda.org/schema/zeebe/1.0" xmlns:modeler="http://camunda.org/schema/modeler/1.0" xmlns:camunda="http://camunda.org/schema/1.0/bpmn" id="Definitions_1" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.19.0" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="8.4.0" camunda:diagramRelationId="8c789d33-bba0-4505-a6f3-15904332cc2d">
+  <bpmn:process id="regressionProcess" isExecutable="true">
+    <bpmn:startEvent id="StartEvent_1">
+      <bpmn:outgoing>Flow_036oth8</bpmn:outgoing>
+    </bpmn:startEvent>
+    <bpmn:sequenceFlow id="Flow_036oth8" sourceRef="StartEvent_1" targetRef="send-email" />
+    <bpmn:serviceTask id="send-email" name="Send mail" zeebe:modelerTemplate="io.camunda.connectors.SendGrid.v2" zeebe:modelerTemplateVersion="2" zeebe:modelerTemplateIcon="data:image/svg+xml;utf8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Cpath%20d%3D%22M0.285706%205.40847H5.43837V10.5611H0.285706V5.40847Z%22%20fill%3D%22white%22%2F%3E%0A%3Cpath%20d%3D%22M0.285706%205.40847H5.43837V10.5611H0.285706V5.40847Z%22%20fill%3D%22%2399E1F4%22%2F%3E%0A%3Cpath%20d%3D%22M5.43837%2010.5611L10.5611%2010.5616V15.6844H5.43837V10.5611Z%22%20fill%3D%22white%22%2F%3E%0A%3Cpath%20d%3D%22M5.43837%2010.5611L10.5611%2010.5616V15.6844H5.43837V10.5611Z%22%20fill%3D%22%2399E1F4%22%2F%3E%0A%3Cpath%20d%3D%22M0.285706%2015.6846L5.43837%2015.6844V15.7143H0.285706V15.6846ZM0.285706%2010.5619H5.43837V15.6844L0.285706%2015.6846V10.5619Z%22%20fill%3D%22%231A82E2%22%2F%3E%0A%3Cpath%20d%3D%22M5.43837%200.285706H10.5611V5.40847H5.43837V0.285706ZM10.5616%205.43837H15.7143V10.5611H10.5616V5.43837Z%22%20fill%3D%22%2300B3E3%22%2F%3E%0A%3Cpath%20d%3D%22M5.43837%2010.5611L10.5611%2010.5616V5.40847H5.43837V10.5611Z%22%20fill%3D%22%23009DD9%22%2F%3E%0A%3Cpath%20d%3D%22M10.5611%200.285706H15.7143V5.40847H10.5611V0.285706Z%22%20fill%3D%22%231A82E2%22%2F%3E%0A%3Cpath%20d%3D%22M10.5611%205.40847H15.7143V5.43837H10.5616L10.5611%205.40847Z%22%20fill%3D%22%231A82E2%22%2F%3E%0A%3C%2Fsvg%3E">
+      <bpmn:extensionElements>
+        <zeebe:taskDefinition type="io.camunda:sendgrid:1" />
+        <zeebe:ioMapping>
+          <zeebe:input source="=key" target="apiKey" />
+          <zeebe:input source="Me" target="from.name" />
+          <zeebe:input source="me@camunda.com" target="from.email" />
+          <zeebe:input source="&#34;You&#34;" target="to.name" />
+          <zeebe:input source="you@camunda.com" target="to.email" />
+          <zeebe:input source="mail" target="unMappedFieldNotUseInModel.mailType" />
+          <zeebe:input source="Hello" target="content.subject" />
+          <zeebe:input source="text/plain" target="content.type" />
+          <zeebe:input source="This &#34;is a&#34; test." target="content.value" />
+        </zeebe:ioMapping>
+        <zeebe:taskHeaders />
+      </bpmn:extensionElements>
+      <bpmn:incoming>Flow_036oth8</bpmn:incoming>
+      <bpmn:outgoing>Flow_1wxg9fx</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:endEvent id="Event_1pddplp">
+      <bpmn:incoming>Flow_1wxg9fx</bpmn:incoming>
+    </bpmn:endEvent>
+    <bpmn:sequenceFlow id="Flow_1wxg9fx" sourceRef="send-email" targetRef="Event_1pddplp" />
+  </bpmn:process>
+  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
+    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="regressionProcess">
+      <bpmndi:BPMNShape id="_BPMNShape_StartEvent_2" bpmnElement="StartEvent_1">
+        <dc:Bounds x="150" y="100" width="36" height="36" />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Activity_1er7c60_di" bpmnElement="send-email">
+        <dc:Bounds x="240" y="78" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Event_1pddplp_di" bpmnElement="Event_1pddplp">
+        <dc:Bounds x="402" y="100" width="36" height="36" />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNEdge id="Flow_036oth8_di" bpmnElement="Flow_036oth8">
+        <di:waypoint x="186" y="118" />
+        <di:waypoint x="240" y="118" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_1wxg9fx_di" bpmnElement="Flow_1wxg9fx">
+        <di:waypoint x="340" y="118" />
+        <di:waypoint x="402" y="118" />
+      </bpmndi:BPMNEdge>
+    </bpmndi:BPMNPlane>
+  </bpmndi:BPMNDiagram>
+</bpmn:definitions>

